connectors/src/connectors/confluence/temporal/client.ts

```diff
@@ -0,0 +1,78 @@
+import type { ModelId } from "@dust-tt/types";
+import type { WorkflowHandle } from "@temporalio/client";
+import { WorkflowNotFoundError } from "@temporalio/client";
+
+import { QUEUE_NAME } from "@connectors/connectors/confluence/temporal/config";
+import { makeConfluenceFullSyncWorkflowId } from "@connectors/connectors/confluence/temporal/utils";
+import { confluenceFullSyncWorkflow } from "@connectors/connectors/confluence/temporal/workflows";
+import { dataSourceConfigFromConnector } from "@connectors/lib/api/data_source_config";
+import { Connector } from "@connectors/lib/models";
+import { getTemporalClient } from "@connectors/lib/temporal";
+import logger from "@connectors/logger/logger";
+
+async function isConfluenceFullSyncAlreadyRunning(connector: Connector) {
+  const client = await getTemporalClient();
+
+  const handle: WorkflowHandle<typeof confluenceFullSyncWorkflow> =
+    client.workflow.getHandle(makeConfluenceFullSyncWorkflowId(connector.id));
+
+  try {
+    const executionDescription = await handle.describe();
+
+    return executionDescription.status.name === "RUNNING";
+  } catch (err) {
+    if (err instanceof WorkflowNotFoundError) {
+      return false;
+    }
+    throw err;
+  }
+}
+
+export async function launchConfluenceFullSyncWorkflow(
```

Comment:
I'll move this logic to https://typescript.temporal.io/api/classes/client.WorkflowClient#signalwithstart so we can better handle when user select in a short time frames many spaces.